src/frontend/Komponenter/Behandling/Vilkårresultat/utils.ts

```diff
@@ -32,3 +35,50 @@ export const summerVilkårsresultat = (
         return acc;
     }, {} as Record<Vilkårsresultat, number>);
 };
+
+export const eksistererVilkårsResultat = (
+    vilkårstypeTilResultat: Record<VilkårType, [Vilkårsresultat]>,
+    resultat: Vilkårsresultat
+): boolean => {
+    // @ts-ignore
+    return [].concat(...Object.values(vilkårstypeTilResultat)).includes(resultat);
+};
+
+export const sorterUtInngangsvilkår = (vilkår: IVilkår): IVurdering[] => {
+    return vilkår.vurderinger.filter((v) => v.vilkårType in InngangsvilkårType);
+};
+
+export const sorterUtAktivitetsVilkår = (vilkår: IVilkår): IVurdering[] => {
+    return vilkår.vurderinger.filter((v) => v.vilkårType in AktivitetsvilkårType);
+};
+
+export const sorterUtTidligereVedtaksvilkår = (vilkår: IVilkår): IVurdering[] => {
+    return vilkår.vurderinger.filter((v) => v.vilkårType in TidligereVedtaksperioderType);
+};
+
+export const erAlleVilkårOppfylt = (vilkår: IVilkår): boolean => {
+    const alleOppfyltBortsettFraAleneomsorg = vilkår.vurderinger.every((vurdering: IVurdering) => {
+        if (vurdering.vilkårType !== InngangsvilkårType.ALENEOMSORG) {
+            return vurdering.resultat === Vilkårsresultat.OPPFYLT;
+        } else {
+            return true;
+        }
+    });
+
+    const aleneomsorgOppfylt = vilkår.vurderinger
+        .filter((vurdering) => vurdering.vilkårType === InngangsvilkårType.ALENEOMSORG)
+        .some((vurdering) => vurdering.resultat === Vilkårsresultat.OPPFYLT);
+
+    return alleOppfyltBortsettFraAleneomsorg && aleneomsorgOppfylt;
+};
+
+export const eksistererIkkeOppfyltVilkår = (vilkår: IVilkår): boolean => {
+    const inngangsvilkår = sorterUtInngangsvilkår(vilkår);
+    const aktivitetsvilkår = sorterUtAktivitetsVilkår(vilkår);
+    const inngangsvvilkårTilResultat = mapVilkårtypeTilResultat(inngangsvilkår);
+    const aktivitetsvilkårTilResultat = mapVilkårtypeTilResultat(aktivitetsvilkår);
+    return (
+        eksistererVilkårsResultat(inngangsvvilkårTilResultat, Vilkårsresultat.IKKE_OPPFYLT) ||
+        eksistererVilkårsResultat(aktivitetsvilkårTilResultat, Vilkårsresultat.IKKE_OPPFYLT)
```

Comment:
Hvorfor splitter vi opp aktivitetsvilkår og inngangsvilkår her - hvorfor er ikke tidligere-vedtaksvilkår interessante? 🤔

Reply:
Og det er faktisk greit om ett av flere aleneomsorg ikke er oppfylt?

Reply:
Med det siste mener jeg => aleneomsorg med to barn -> én oppfylt og én ikke oppfylt betyr i praksis at vilkåret er oppfylt ✅